elasticapm/metrics/base_metrics.py

```diff
@@ -202,33 +209,42 @@ def collect(self):
         samples = defaultdict(dict)
         if self._counters:
             # iterate over a copy of the dict to avoid threading issues, see #717
-            for (name, labels), c in compat.iteritems(self._counters.copy()):
-                if c is not noop_metric:
-                    val = c.val
-                    if val or not c.reset_on_collect:
-                        samples[labels].update({name: {"value": val}})
-                    if c.reset_on_collect:
-                        c.reset()
+            for (name, labels), counter in compat.iteritems(self._counters.copy()):
+                if counter is not noop_metric:
+                    val = counter.val
+                    if val or not counter.reset_on_collect:
+                        samples[labels].update({name: {"value": val, "type": "counter"}})
```

Comment:
I think we should omit `"type": "counter"` if the counter is resetting; just omit "type" altogether for now. Counter metrics are expected to be monotonically increasing.